### PR TITLE
DDLs for tables

### PR DESCRIPTION
**Pull request by dumbapple**

### CODE REVIEW

File: EventFinder-DDLs.sql [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    Age INT NOT NULL
+);
+
+CREATE TABLE User (
+    UserID INT PRIMARY KEY AUTO_INCREMENT,
+    FirstName VARCHAR(50) NOT NULL,
+    LastName VARCHAR(50) NOT NULL,
+    DateOfBirth DATE NOT NULL,
+    Gender VARCHAR(50) NOT NULL,
+    FOREIGN KEY (DateOfBirth) REFERENCES DoB(DateOfBirth) ON DELETE CASCADE ON UPDATE CASCADE
```
Comment (Bosco28) [inline EventFinder-DDLs.sql]: Let's do `ON DELETE NO ACTION` instead of `ON DELETE CASCADE`

File: EventFinder-DDLs.sql [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+CREATE TABLE CreditCard (
+    CardNumber VARCHAR(16) PRIMARY KEY,
+    ExpiryDate DATE NOT NULL,
+    HolderName VARCHAR(100) NOT NULL,
+    CVC INT NOT NULL
+);
+
+CREATE TABLE Organizer (
+    OrganizerUserID INT PRIMARY KEY AUTO_INCREMENT,
```
Comment (Bosco28) [inline EventFinder-DDLs.sql]: This is also a foreign key referring to the user table. Therefore we don't need `AUTO_INCREMENT` here.

File: EventFinder-DDLs.sql [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    EventID INT PRIMARY KEY AUTO_INCREMENT,
+    Title VARCHAR(200) NOT NULL,
+    StartDate TIMESTAMP NOT NULL,
+    EndDate TIMESTAMP NOT NULL,
+    Description VARCHAR(200),
+    LocationAddress VARCHAR(200),
+    OrganizerUserID INT,
+    FOREIGN KEY (LocationAddress) REFERENCES Location(Address) ON DELETE NO ACTION,
+    FOREIGN KEY (OrganizerUserID) REFERENCES User(UserID) ON DELETE SET NULL,
+    FOREIGN KEY (OrganizerUserID) REFERENCES Organizer(OrganizerUserID) ON DELETE CASCADE
```
Comment (Bosco28) [inline EventFinder-DDLs.sql]: Let's do `ON DELETE NO ACTION` here

**Review by Bosco28 — CHANGES_REQUESTED** [review]
Can you also add Inder's INSERT statements to this file? Thanks!

File: EventFinder-DDLs.sql [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+CREATE TABLE Staff (
+    StaffID INT PRIMARY KEY AUTO_INCREMENT,
+    Name VARCHAR(100) NOT NULL,
+    PayRate INTEGER NOT NULL
+);
+
+CREATE TABLE Location (
+    Address VARCHAR(200) PRIMARY KEY,
+    Name VARCHAR(200) NOT NULL
+);
```
Comment (Bosco28) [inline EventFinder-DDLs.sql]: Move `Location` to come before `Event` otherwise there will be a foreign key error.

File: EventFinder-DDLs.sql [hunk 5]
```diff
@@ -0,0 +1,190 @@
+CREATE TABLE DoB (
```
Comment (Bosco28) [inline EventFinder-DDLs.sql]: Can you also add drop table queries here? Thanks!

**Review by Bosco28 — CHANGES_REQUESTED** [review]

**Comment by dumbapple** [comment]
Btw the insert statements have to be in the order they appear in or it might fail due to foreign key constraints

File: EventFinder-DDLs.sql [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+
+CREATE TABLE Organizer (
+    OrganizerUserID INT PRIMARY KEY,
+    OrganizerContactInfo VARCHAR(200) NOT NULL,
+	FOREIGN KEY (OrganizerUserID) REFERENCES User(UserID) ON DELETE CASCADE
+);
+
+CREATE TABLE Location (
+    Address VARCHAR(200) PRIMARY KEY,
+    Name VARCHAR(200) NOT NULL
```
Comment (Bosco28) [inline EventFinder-DDLs.sql]: Location does not have name anymore. Let's just completely remove the location table and all its foreign keys.

File: EventFinder-DDLs.sql [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+DROP TABLE Sponsor;
+DROP TABLE EventType;
+DROP TABLE WorkOn;
+DROP TABLE EventHasType;
+DROP TABLE InterestedIn;
+DROP TABLE Friend;
+DROP TABLE PayWith;
+DROP TABLE Participate;
+
+CREATE TABLE DoB (
```
Comment (Bosco28) [inline EventFinder-DDLs.sql]: Let's remove the DoB table. We don't really need it anymore.

**Review by Bosco28 — CHANGES_REQUESTED** [review]

File: EventFinder-DDLs.sql [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+INSERT INTO User VALUES (51, 'JOHN', 'SMITH', '1991-05-23', 'Male');
+INSERT INTO User VALUES (52, 'PETER', 'PARKER', '1991-06-02','Male');
+INSERT INTO User VALUES (53, 'Tony', 'Stark', '1972-12-10','Male');
+INSERT INTO User VALUES (54, 'Black', 'Widow', '1982-10-06','Female');
+INSERT INTO User VALUES (55, 'Steve', 'Rogers', '1933-04-25','Male');
+INSERT INTO Event VALUES(0, 'Comedy Club', '2020-04-15', '2020-04-17','Full of laughs','Beatty Street, Vancouver', NULL, NULL);
+INSERT INTO Event VALUES(1, 'UFC Fight','2020-05-03','2020-05-03', 'McGreggor Vs Khalid', 'Gastown', 55, "55@eventfinder.com" );
+INSERT INTO Event VALUES(2, 'Circus', '2020-04-23','2020-05-03','Cirque de Solil', 'Granville Street', 53, "53@eventfinder.com" );
+INSERT INTO Event VALUES(3, 'Play', '2020-05-24','2020-06-07','Theatrical Play','Mainland Street', Null, Null );
+INSERT INTO Event VALUES(4, 'Drinking Challenge', '2020-06-07','2020-06-09', 'Beer Drinking Competition','Gastown',NULL, NULL );
```
Comment (Bosco28) [inline EventFinder-DDLs.sql]: Let's not allow an event to not have an organizer.

File: EventFinder-DDLs.sql [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+
+CREATE TABLE Event (
+    EventID INT PRIMARY KEY AUTO_INCREMENT,
+    Title VARCHAR(200) NOT NULL,
+    StartDate TIMESTAMP NOT NULL,
+    EndDate TIMESTAMP NOT NULL,
+    Description VARCHAR(200),
+    LocationAddress VARCHAR(200),
+    OrganizerUserID INT,
+	OrganizerContactInfo VARCHAR(200),
```
Comment (Bosco28) [inline EventFinder-DDLs.sql:39]: Let's make these not null

**Review by Bosco28 — CHANGES_REQUESTED** [review]

**Pull request merged**